File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        PriorityQueue<Integer> hq = new PriorityQueue<>();
        
        for (int i=0; i<scoville.length; i++) {
            hq.add(scoville[i]);
        }
        
        while (true) {
            
            if (hq.size() < 2 && hq.peek() < K) {
                answer = -1;
                break;
            }
            
            if (hq.peek() >= K) {
                break;
            }
            
            int min1 = hq.poll();
            int min2 = hq.poll();
            
            hq.offer(min1 + (min2 * 2));
            answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    
    heapq.heapify(scoville)
    
    while scoville[0] < K:
        if len(scoville) < 2:
            return -1
        
        a = heapq.heappop(scoville)
        b = heapq.heappop(scoville)
        
        heapq.heappush(scoville, a + (b*2))
        answer += 1
        
    return answer